src/utils/DependsOnManager.ts

```diff
@@ -28,7 +28,7 @@ export class DependsOnManager {
   private parentFacetRef: IDependsOnCompatibleFacet;
 
   constructor(private facet: IDependentFacet) {
-    this.facet.ref.bind.onRootElement(QueryEvents.buildingQuery, () => this.handleBuildingQuery());
+    $$(this.facet.ref.searchInterface.element).on(QueryEvents.buildingQuery, () => this.handleBuildingQuery());
```

Comment:
There are multiple solutions to this PR, such as preventing `DynamicFacet` from toggling out `coveo-hidden` or deferring the `disable()` call.

However, I found this solution to be the least risky and most elegant since even though it "worked" for the `Facet`, one facet was hidden+disabled while the other was only hidden.

Reply:
Nice!